topics/single-cell/tutorials/pseudobulk-analysis/tutorial.md

```diff
@@ -0,0 +1,503 @@
+> >    The **Samples Metadata** is another file that will serve as an input for the edgeR tool.
+> >
+> {: .solution}
+>
+{: .question}
+
+## Sanitation Steps - Part 1
+
+The next steps will help you refine your data for easier handling. We will use standard galaxy tools, like: [Replace Text](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1), [Remove columns](https://toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) and [Text reformatting](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1). 
+
+> <hands-on-title> Replace Text: Count Matrix</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `count_matrix` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+{: .hands_on}
+
+
+> <hands-on-title> Replace Text: Sample Metadata</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `samples_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+{: .hands_on}
+
+> <hands-on-title> Remove Columns: Gene Metadata </hands-on-title>
+>
+> 1. {% tool [Remove columns](toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) %} with the following parameters:
+>    - {% icon param-file %} *"Tabular file"*: `genes_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Select Columns"*:
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `start`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `end`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `width`
+>
+{: .hands_on}
+
+> <hands-on-title> Replace Text: Creating Factor File </hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `outfile` (output of **Replace Text: Sample Metadata Step** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"in column"*: `c2`
+>            - *"Find pattern"*: `^([0-9])(.+)`
+>            - *"Replace with"*: `GG_\\1\\2`
+>
+{: .hands_on}
```

Comment:
here are four hand-on boxes without text in between. You can just merge them into one big box with 4 steps. 

Or add some text in between explaining what the steps do or discussing the output etc of course.